alexandria/core/views.py

```diff
@@ -259,3 +263,36 @@ def download(self, request, pk=None):
                 obj.content.file.file, as_attachment=False, filename=obj.name
             )
         raise PermissionDenied("For downloading a file use the presigned download URL.")
+
+    @action(methods=["get"], detail=False)
+    def search(self, request, **kwargs):
+        query = request.query_params.get("filter[query]")
+        if not query:
+            raise ValidationError(_('Specifying a "query" filter is mandatory!'))
+
+        queryset = self.get_queryset()
+
+        search_queries = []
+        for lang, desc in settings.LANGUAGES:
+            config = settings.ISO_639_TO_PSQL_SEARCH_CONFIG.get(lang, "simple")
+            search_queries.append(
+                Q(language=lang)
+                & Q(
+                    content_vector=SearchQuery(
+                        query, config=config, search_type="phrase"
+                    )
+                )
+            )
+
+        search_query = reduce(or_, search_queries)
+
+        latest_files = models.File.objects.filter(
+            document=OuterRef("document_id"), variant=models.File.Variant.ORIGINAL
+        ).order_by("-created_at")
+        queryset = queryset.filter(pk=Subquery(latest_files.values("pk")[:1])).filter(
+            search_query
+        )
+
+        serializer = self.get_serializer(queryset, many=True)
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=HTTP_200_OK, headers=headers)
```

Comment:
Code functionally LGTM, but I suggest keeping it in the "traditional" `FilterSet` structure.
This implies adding a new (readonly) viewset, with a new serializer that also exposes the headline / search rank as well.

Reply:
Changed to a filter method. Can be refactored to new viewset when integration additional full text search features.

Reply:
I would suggest extending it to a new viewset (and endpoint) now, as it will be a breaking change if you do it later. And I think the headline/rank will be added relatively quickly

Reply:
Ok ive created a view for it.